### PR TITLE
refactor: refactored logic for sending marketing email

### PR DESCRIPTION
**Pull request by aayushchugh**
Check postman for request and response example

### Checklist:

-   [x] This pull request follow our contribution guidelines
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (postman and wiki)
-   [ ] I have written tests for the code
-   [ ] I have updated .env.example file for new .env variables

image

### CODE REVIEW

File: src/controllers/marketingEmail.controller.ts [hunk 1]
```diff
@@ -1,61 +1,79 @@
...
+		/**
+		 * Send email to specific users
+		 */
+		if (specificUsers && specificUsers.length > 0) {
+			// get specific users from DB
+			const users = await Promise.all(
+				specificUsers.map((userId) => findUserByIdService(userId)),
+			);
+
+			users.forEach(async (user) => user && (await sendEmail(user.email, subject, html)));
```
Comment (tharun634) [inline src/controllers/marketingEmail.controller.ts]: Do we have to see if the `user` has `receiveMarketingEmails: true`? Or is it assured that `specificUsers` will only be a subset of `user`'s who has `receiveMarketingEmails: true`?

Comment (aayushchugh) [reply src/controllers/marketingEmail.controller.ts]: Oh, I forgot to add validation for that

**Review by BitOfAByte — APPROVED** [review]

**Review by is-it-ayush — APPROVED** [review]
Approved! Merging this for a short refactor later.

**Review by tharun634 — APPROVED** [review]
LGTM!

**Pull request merged**